src/activity/activity-pool.ts

```diff
@@ -97,8 +97,17 @@ export class ActivityPool {
     this.events.emit("destroyed", activity.getDto());
```

Comment:
this event should be emitted from the function in `createFactoryPool` because it will not be emitted here if the activity is destroyed during the `drainAndClear` process